File: Data Structures & Algorithms/max-water-container/submission-0.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(nums) {
        let max = 0;
        for(let i = 0; i<nums.length; i++){
            for(let j = i+1; j<nums.length;j++){
               let curr_max  = Math.min(nums[i], nums[j]) * (j - i)
                max = Math.max(curr_max, max)
            }
        }
        return max
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        res = nums[0]
        i = 0
        j = len(nums) - 1
        while i <= j:
            if nums[i] < nums[j]:
                res = min(res, nums[i])
                break
            mid = (i+j) // 2
            res = min(res, nums[mid])
            if nums[mid] >= nums[i]:
                i = mid + 1
            else:
                j = mid - 1
        return res